### ISSUE TITLE
Maven metadata expiration time should be configurable in application.yaml

### ISSUE THREAD

**Issue by fuss86** [strongbox/strongbox#1335]
# Task Description

Currently `maven-metadata.xml` is considered as expired (see [Maven2LayoutProvider ](https://github.com/strongbox/strongbox/blob/master/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/Maven2LayoutProvider.java#L101)) after 10 seconds of last modification date. This value should be configurable in [application.yaml](https://github.com/strongbox/strongbox/blob/master/strongbox-commons/src/main/resources/application.yaml) file.

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @sbespalov
  * @fuss86

**Comment by ajaybhat** [comment]
I can do this. Shall I name the value as `maven.metadata.expirationtime`?

**Comment by carlspring** [comment]
@ajaybhat ,

I've updated the description. Please, feel free to proceed! :)

**Comment by TylerHaze** [comment]
Looks like not much has happened on this so I'd be willing to pick it up! I jumped into chat, trying to understand a few things. Where does the `maven.metadata.expiration.time` come from? Is the YAML creating that? If not, where are these coming from? Thanks.

**Comment by fuss86** [comment]
@TylerHaze , yeah, pick this task up, please.

We do store properties in [application.yaml](https://github.com/strongbox/strongbox/blob/master/strongbox-commons/src/main/resources/application.yaml). You should put there a global key `strongbox.maven.metadata.expiration.timeSeconds` and value `3600`.

Also, as @carlspring mentioned, there should also be an option to configure it per repository.

For now, you can start from global setting.

Thanks !

---

@ajaybhat, please let us know if you are still working on this task, because it doesn't look so :)

Thanks !

**Comment by TylerHaze** [comment]
Ok, so I think for the I just do something like...

```
strongbox:
  ...{bunch of other config code here}
  maven:
    metadata:
      expiration:
        timeSeconds: 3600
```
But how do you actually get this code into Java? Like, if I want to get that value `3600` from my application.yaml, what is it that does that?

**Comment by fuss86** [comment]
Search for `@Value` annotation around in the source code :)

**Comment by sbespalov** [comment]
@fuss86 @carlspring this property probably better to be placed in `strongbox.yaml` as all storages and repositories configurations are there.

**Comment by carlspring** [comment]
You are absolutely right, @sbespalov. I hadn't realized the description was wrong. I've corrected it now.

**Comment by fuss86** [comment]
@sbespalov yeah, since it is going to be set per repository, it's better to be placed there, thanks 👍

**Comment by fuss86** [comment]
@TylerHaze are you still working on this task ?

**Comment by Nishant-sehgal** [comment]
@TylerHaze are you still working on it..If not will like to give an attempt on it.
CC @fuss86 Please tell if i can pick this up.

**Comment by carlspring** [comment]
@Nishant-sehgal : Please, feel free to pick it up!

**Comment by Nishant-sehgal** [comment]
@carlspring have some questions with the approach
was adding below as the property in strongbox.yaml.

configuration:
  XXX
  maven:
    metadata:
      expiration:
        timeSeconds: 3600

1. can you check if it should come under configuration directly?
2. i will need to map it to a POJO with name Maven which contains Metadata as reference inside it and expiration in metadata and so on.Is this right?
3. use configuration manager to read the value in  Maven2LayoutProvider and use expirytime there? 

Can't we have below property. It would be much simpler and shorter. what say?

  maven:
    metadata:
      expirationSeconds: 3600

Please correct me if i am going in wrong direction.

**Comment by carlspring** [comment]
@fuss86: Would you mind giving @nishant-sehgal some hints and tips?

@nishant-sehgal: It would probably be best to join our [chat channel](https://chat.carlspring.org/].

**Comment by steve-todorov** [comment]
Just FYI guys - this task has a new relationship with #1460

// cc @cbrianhill

**Comment by fuss86** [comment]
> Can't we have below property. It would be much simpler and shorter.

Indeed it looks nicer. Thanks for your suggestion. Please us it.

>  1. can you check if it should come under configuration directly?

Thanks for pointing this out. Indeed it's not a good option. We should create some generic interface (like `org.carlspring.strongbox.yaml.repository.RepositoryConfiguration` but for global providers configuration) and create implementations per repository providers if needed. `Maven` case will be the first one.

> 2. i will need to map it to a POJO with name Maven which contains Metadata as 
   reference inside it and expiration in metadata and so on.Is this right?

No, it is not right. Please annotate classes by `@JsonRootName` (or use other `jackson` annotations if needed).

3. use configuration manager to read the value in  Maven2LayoutProvider and use 
   expirytime there? 

Yes, that's correct :)

**Comment by Nishant-sehgal** [comment]
@fuss86  if i got it right inside strongbox.yaml you are suggesting something like this:

```yaml
configuration:
  XXX {other props}
  repository:
    repositoryConfiguration:
     expirationSeconds: 3600 {can have global expiry here }
     maven:
       metadata:
         expirationSeconds: 3600
```

**Comment by carlspring** [comment]
@Nishant-sehgal 

What is the status of this task?

**Comment by carlspring** [comment]
This task is now once again up for grabs.
